### PR TITLE
SplitNCigar reads walker refactor 1864

### PR DESCRIPTION
**Pull request by jamesemery**
I refactored SplitNCigar reads to use the ReadWalker interface

### CODE REVIEW

**Comment by akiezun** [comment]
for @lbergelson to review

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/SplitNCigarReads.java [hunk 1]
```diff
@@ -91,46 +90,43 @@
...
+        SAMFileHeader header = getHeaderForSAMWriter();
+        rnaReadTransform = REFACTOR_NDN_CIGAR_READS ? new NDNCigarReadTransformer() : ReadTransformer.identity();
         try {
-            final IndexedFastaSequenceFile referenceReader = new CachingIndexedFastaSequenceFile(REFERENCE_SEQUENCE);
-            GenomeLocParser genomeLocParser= new GenomeLocParser(referenceReader.getSequenceDictionary());
-            overhangManager = new OverhangFixingManager(outputHeader, outputWriter, genomeLocParser, referenceReader, MAX_RECORDS_IN_MEMORY, MAX_MISMATCHES_IN_OVERHANG, MAX_BASES_TO_CLIP, doNotFixOverhangs);
-            return outputWriter;
+            referenceReader = new CachingIndexedFastaSequenceFile(referenceArguments.getReferenceFile());
+            GenomeLocParser genomeLocParser = new GenomeLocParser(getBestAvailableSequenceDictionary());
+            outputWriter = new SAMFileGATKReadWriter(ReadUtils.createCommonSAMWriter(OUTPUT, referenceArguments.getReferenceFile(), header, false, false, false));
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/SplitNCigarReads.java]: So `ReadWalker` has a method "createSAMWriter" that's probably better to use than creating one manually.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/SplitNCigarReads.java [hunk 2]
```diff
@@ -91,46 +90,43 @@
...
-    private SAMFileGATKReadWriter initialize(final SamReader in) {
-        final SAMFileHeader outputHeader = ReadUtils.cloneSAMFileHeader(in.getFileHeader());
-        //preSorted is set to false for this writer since ReadCoordinateComparator doesn't match HTSJDK's when an unmapped read has a mapped mate.
-        //Also, we don't want to assume that reads will be written in order by the manager because in deep, deep pileups it won't work
-        final SAMFileGATKReadWriter outputWriter = new SAMFileGATKReadWriter(ReadUtils.createCommonSAMWriter(OUTPUT, REFERENCE_SEQUENCE, outputHeader, false, false, false));
-
+    
+    @Override
+    public void onTraversalStart() {
+        final IndexedFastaSequenceFile referenceReader;
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/SplitNCigarReads.java]: I think this should probably be moved to be a class field and then be closed in the tool shutdown.

**Comment by jamesemery** [comment]
@lbergelson addressed the comments and added a test, please rereview

**Comment by lbergelson** [comment]
👍

**Pull request merged**